tests/db/test_base.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import pytest
+import sqlite3
+
+from monkeytype.db.base import CallTraceStoreLogger
+from monkeytype.db.sqlite import (
+    create_call_trace_table,
+    SQLiteStore,
+)
+from monkeytype.tracing import trace_calls
+from _pytest.monkeypatch import MonkeyPatch
+
+
+def func(a, b):
+    pass
+
+
+def func2(a, b):
+    pass
+
+
+def func3(a, b):
+    pass
+
+
+@pytest.fixture
+def logger() -> CallTraceStoreLogger:
+    conn = sqlite3.connect(':memory:')
+    create_call_trace_table(conn)
+    return CallTraceStoreLogger(SQLiteStore(conn))
+
+
+def test_round_trip(logger):
+    from types import ModuleType
+    module = ModuleType('__main__')
+    module.func = func3
+    MonkeyPatch().setattr(module.func, '__module__', '__main__', raising=False)
```

Comment:
The correct way to use `MonkeyPatch` is not by importing and instantiating it yourself (the clue is that you have to import it from `_pytest`; anything that's supposed to be imported will be in the public `pytest` namespace). It's a fixture, you can request a pre-built instance by just adding `monkeypatch` to the test signature, and then do `monkeypatch.setattr(...)`.

Personally I'd probably just use `from unittest.mock import patch` and then `with patch.object(...)` instead, since that's what we use elsewhere in the tests. But I don't feel strongly, using the `monkeypatch` fixture is OK too.

Reply:
Thanks for the advice on pytest, I've never used monkeypatch before.

The changes you suggested are updated.